src/common/reactor.cc

```diff
@@ -262,12 +264,12 @@ struct SyncImpl : public Reactor::Impl {
         }
 
         static Polling::Tag encodeTag(const Reactor::Key& key, uint64_t value) {
-            auto index = key.data();
+          //  auto index = key.data();
             // The reason why we are using the most significant bits to encode
             // the index of the handler is that in the fast path, we won't need
             // to shift the value to retrieve the fd if there is only one handler as
             // all the bits will already be set to 0.
-            auto encodedValue = (index << HandlerShift) | value;
+           // auto encodedValue = (index << HandlerShift) | value;
```

Comment:
You are actually god damn right and guess what, you discovered a severe bug. I was actually lucky because I'm registering only one handler in the Reactor system but part of this code should be fixed, good catch !